compiler/test/dotty/tools/backend/jvm/ArrayApplyOptTest.scala

```diff
@@ -22,18 +22,18 @@ class ArrayApplyOptTest extends DottyBytecodeTest {
     test("Array[Char]()", newArray0Opcodes(T_CHAR))
     test("Array[T]()", newArray0Opcodes(T_INT))
 
-    test("IArray[String]()", List(Op(ICONST_0), TypeOp(ANEWARRAY, "java/lang/String"), TypeOp(CHECKCAST, "[Ljava/lang/String;"), Op(POP), Op(RETURN)))
-    test("IArray[Unit]()", List(Op(ICONST_0), TypeOp(ANEWARRAY, "scala/runtime/BoxedUnit"), TypeOp(CHECKCAST, "[Lscala/runtime/BoxedUnit;"), Op(POP), Op(RETURN)))
-    test("IArray[Object]()", List(Op(ICONST_0), TypeOp(ANEWARRAY, "java/lang/Object"), TypeOp(CHECKCAST, "[Ljava/lang/Object;"), Op(POP), Op(RETURN)))
-    test("IArray[Boolean]()", newArray0Opcodes(T_BOOLEAN, TypeOp(CHECKCAST, "[Z") :: Nil))
-    test("IArray[Byte]()", newArray0Opcodes(T_BYTE, TypeOp(CHECKCAST, "[B") :: Nil))
-    test("IArray[Short]()", newArray0Opcodes(T_SHORT, TypeOp(CHECKCAST, "[S") :: Nil))
-    test("IArray[Int]()", newArray0Opcodes(T_INT, TypeOp(CHECKCAST, "[I") :: Nil))
-    test("IArray[Long]()", newArray0Opcodes(T_LONG, TypeOp(CHECKCAST, "[J") :: Nil))
-    test("IArray[Float]()", newArray0Opcodes(T_FLOAT, TypeOp(CHECKCAST, "[F") :: Nil))
-    test("IArray[Double]()", newArray0Opcodes(T_DOUBLE, TypeOp(CHECKCAST, "[D") :: Nil))
-    test("IArray[Char]()", newArray0Opcodes(T_CHAR, TypeOp(CHECKCAST, "[C") :: Nil))
-    test("IArray[T]()", newArray0Opcodes(T_INT, TypeOp(CHECKCAST, "[I") :: Nil))
+    test("IArray[String]()", List(Op(ICONST_0), TypeOp(ANEWARRAY, "java/lang/String"), TypeOp(CHECKCAST, "[Ljava/lang/String;"), VarOp(ASTORE, 1), VarOp(ALOAD, 1), Op(POP), Op(RETURN)))
+    test("IArray[Unit]()", List(Op(ICONST_0), TypeOp(ANEWARRAY, "scala/runtime/BoxedUnit"), TypeOp(CHECKCAST, "[Lscala/runtime/BoxedUnit;"), VarOp(ASTORE, 1), VarOp(ALOAD, 1), Op(POP), Op(RETURN)))
+    test("IArray[Object]()", List(Op(ICONST_0), TypeOp(ANEWARRAY, "java/lang/Object"), TypeOp(CHECKCAST, "[Ljava/lang/Object;"), VarOp(ASTORE, 1), VarOp(ALOAD, 1), Op(POP), Op(RETURN)))
+    test("IArray[Boolean]()", newArray0Opcodes(T_BOOLEAN,  TypeOp(CHECKCAST, "[Z") :: VarOp(ASTORE, 1) :: VarOp(ALOAD, 1) :: Nil))
```

Comment:
These two extra operations are completely unnecessary. Maybe it can be handled in [TupleOptimizations.scala](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/transform/TupleOptimizations.scala).

Reply:
Perhaps, this is out of the scope of this PR (I don't have time to investigate). I'd rather revert the `apply` implementations instead.

Reply:
It is better to revert that change than testing a regression.